src/attr/_make.py

```diff
@@ -1558,6 +1613,17 @@ def from_counting_attr(cls, name, ca, type=None):
             **inst_dict
         )
 
+    # Don't use attr.evolve since fields(Attribute) doesn't work
+    def _evolve(self, **changes):
```

Comment:
This might seem pedantic, but what you do here doesn't mirror our evolve but our assoc. Just call it what it does: `copy_then_set` or something.

Reply:
Renamed to match the semantics. As `Attribute` isn't an `attrs` class  `assoc` can't be used for the same reason, it depends on `fields`.  I'd avoided the `assoc` name as the function appears to be deprecated and, in this class, the functions are effectively equivalent.